docs/ancestry.md

```diff
@@ -1264,13 +1265,86 @@ print(tree.tmrca(0,1))
 
 ### Selective sweeps
 
-```{eval-rst}
-.. todo:: Document the selective sweep models.
+Msprime provides the option to perform coalescent approximations
+to selective sweeps, in which a beneficial mutation moves through
+the population. This is done through the use of a structured 
+coalescent model in the spirit of 
+[Braverman et al. (1995)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206652/).
+
+Looking backwards in time the population enters a sweep phase where
+the population is split between two classes of selective backgrounds,
+those linked to the beneficial allele, call it {math}`B`, and those not,
+{math}`b`. During this portion of the simulation we track competing
+rates of coalescence and recombination on the {math}`B` and {math}`b`
+backgrounds.
+
+This implementation is reasonable general, althougth there are 
+current limitations (e.g., no change of population size during
+a sweep). The user supplies a final allele frequency and a starting
+allele frequency, between which msprime will simulate a stochastic 
+sweep trajectory according to a conditional diffusion model [(Coop
+and Griffiths, 2004](https://pubmed.ncbi.nlm.nih.gov/15465123/);
+[Kim and Stephan 2002](https://pubmed.ncbi.nlm.nih.gov/11861577/)).
+
+Beyond the start and end frequencies of the sweep trajectory, the user
+must specify the selection coefficient of the beneficial mutation 
+{math}`s` the selective
+advantage of the {math}`B` homozygote over the {math}`b` homozygote
+and {math}`h=0.5`. The position represents the location along
+the chromosome where the beneficial allele occurs.
+ All other parameters can be set as usual. 
+
+As an example, let's perform 50 replicate hard sweep simulations using 
+msprime and then plot mean {meth}`pairwise diversity<tskit.TreeSequence.diversity>` in windows across the simulated
+region. The position is set to the middle of the chromosome, so we
+expect to see the characteristic valley of diversity following our sweeps.
+The key step here is to build a list of models which includes
+a {class}`SweepGenicSelection<.SweepGenicSelection>` model.
```

Comment:
Can use  {class}`.SweepGenicSelection` here because we don't need to specify custom link text